modules/reitit-core/src/reitit/core.cljc

```diff
@@ -334,13 +334,36 @@
 ;; Creating Routers
 ;;
 
+(defn leaf-endpoint?
+  [_ _ _ _ childs]
+  (not (seq childs)))
+
+(defn mk-intermediate-endpoint-predicate
+  "Create predicate that returns truthy for valid endpoints, where the path is a leaf in
+  the route tree, or the route data contains one of the specified keys at top-level or
+  at the first level of nested maps.
```

Comment:
Looking at nested maps can muddle with user specific data, e.g.

```clj
["/topic" {:topic {:name "iot.inbox"}}]
```

Reply:
True. Changed it so that the default is to check for handler defined under http methods.